public/app/plugins/datasource/graphite/configuration/GraphiteDetails.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { PureComponent } from 'react';
+import { DataSourceSettings, SelectableValue } from '@grafana/data';
+import { Button, FormLabel, Select } from '@grafana/ui';
+import { GraphiteOptions, GraphiteType } from '../types';
+
+const graphiteVersions = [
+  { label: '0.9.x', value: '0.9' },
+  { label: '1.0.x', value: '1.0' },
+  { label: '1.1.x', value: '1.1' },
+];
+
+const graphiteTypes = Object.keys(GraphiteType).map((key: string) => ({
+  label: key,
+  value: (GraphiteType as any)[key],
+}));
+
+interface Props {
+  value: DataSourceSettings<GraphiteOptions>;
+  onChange: (value: DataSourceSettings<GraphiteOptions>) => void;
+}
+
+interface State {
+  showMetricTankHelp: boolean;
+}
+
+export class GraphiteDetails extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      showMetricTankHelp: false,
+    };
+  }
+
+  onChangeHandler = (key: keyof GraphiteOptions) => (newValue: SelectableValue) => {
+    const { value, onChange } = this.props;
+    onChange({
+      ...value,
+      jsonData: {
+        ...value.jsonData,
+        [key]: newValue.value,
+      },
+    });
+  };
+
+  render() {
+    const { value } = this.props;
+    const { showMetricTankHelp } = this.state;
+
+    return (
+      <>
+        <h3 className="page-heading">Graphite details</h3>
+        <div className="gf-form-group">
+          <div className="gf-form">
+            <FormLabel tooltip="This option controls what functions are available in the Graphite query editor.">
+              Version
+            </FormLabel>
+            <Select
+              value={graphiteVersions.find(version => version.value === value.jsonData.graphiteVersion)}
+              options={graphiteVersions}
+              width={8}
+              onChange={this.onChangeHandler('graphiteVersion')}
+            />
+          </div>
+          <div className="gf-form-inline">
+            <FormLabel>Type</FormLabel>
+            <Select
+              options={graphiteTypes}
+              value={graphiteTypes.find(type => type.value === value.jsonData.graphiteType)}
+              width={8}
+              onChange={this.onChangeHandler('graphiteType')}
+            />
```

Comment:
On hosted grafana, this switch is not visible but I can't figure out what is hiding it?

Reply:
It's implemented in 6.5